File: 1. Harvard CS50/sentimental-mario-more/mario.py
# TODO
from cs50 import get_int


def main():
    while True:
        height = get_int("Height: ")
        if height > 0 and height < 9:
            break

    for i in range(height):
        spaces = " " * (height - 1 - i)
        hashes_left = "#" * (i + 1)
        hashes_right = "#" * (i + 1)
        print(f"{spaces}{hashes_left}  {hashes_right}")


main()

File: 1. Harvard CS50/mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

int getHeight(void);
void printPyramid(int height);

// Asks a height and returns a pyramid
int main(void)
{
    // TODO: prompt the height of pyramid
    int height = getHeight();

    // TODO: print the pyramid
    printPyramid(height);
}

int getHeight(void)
{
    int height;
    do
    {
        height = get_int("Height: ");
    }
    while (height > 8 || height < 1);
    return height;
}

void printPyramid(int height)
{
    for (int i = 0; i < height; i++)
    {
        // Print the spaces only if the height is above of 1
        for (int space = 0; space < height - 1 - i; space++)
        {
            printf(" ");
        }

        // Print left pyramid
        for (int j = 0; j <= i; j++)
        {
            printf("#");
        }

        printf("  ");

        // Print right pyramid
        for (int k = 0; k <= i; k++)
        {
            printf("#");
        }

        // New column
        printf("\n");
    }
}